packages/proposals

```diff
@@ -0,0 +1,53 @@
+1 PKG_PRIORITY
+
+None of the current scripts use the PKG_PRIORITY variable.
+All but three packages set PKG_PRIORITY="optional". busybox, dbus and systemd
+set PKG_PRIORITY="required".
+The effect of PKG_PRIORITY is unknown.
+
+Proposal:
+Deprecate the PKG_PRIORITY variable
+
+
+2 PKG_SITE, PKG_LONGDESC, PKG_SHORTDESC
+
+None of the current scripts use the PKG_SITE variable.
+create_addon is the only script that uses the PKG_LONGDESC and PKG_SHORTDESC
+variables.
+In other words, the PKG_LONGDESC and PKG_SHORTDESC are only used for
+addons. For other packages, the PKG_SITE, PKG_LONGDESC and PKG_SHORTDESC are
+merely informative and redundant.
+
+Proposal:
+Deprecate PKG_LONGDESC and/or PKG_SHORTDESC for normal packages
+
+
+3 PKG_AUTORECONF
+
+More than 330 packages set PKG_AUTORECONF="no", whereas about 230 packages set
+PKG_AUTORECONF="yes".
+build is the only script that uses the PKG_AUTORECONF variable.
+
+Proposal: 
+Modify build to set PKG_AUTORECONF="no" if PKG_AUTORECONF is not set in
+in package.mk
```

Comment:
I'd actually leave this as-is - if it's an almost 50/50 split then there is no reasonable default, so it should continue to be specified explicitly by each package. On the other hand if 557 packages had "no" and 3 had "yes" then sure, making "no" the default is perhaps something to consider.

Another thing about allowing values to default, is that it "hides" implementation details - you can look at the package.mk and without knowing the defaults for every possible PKG_* variable you don't have the full picture of how the package will be built - there could be variables that are not present in the package.mk which, because of default values, have a very significant outcome on how the package is built. For this reason I would caution against using defaults wherever possible, and always favour being explicit.